### ISSUE TITLE
Code fix for CA1822 to make member static

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#1929]
```c#
using System;

class C
{
    void M()
    {
        Console.WriteLine(); // CA1822: Member M does not access instance data and can be marked as static (Shared in VisualBasic)
    }
}
```
It would be nice to have a code fix to make `M` static.

**Comment by mavasani** [comment]
We had a PR for this, which got closed as providing a safe and performance fix is quite complicated: 
https://github.com/dotnet/roslyn-analyzers/pull/1116

I agree that we should probably provide some kind of a fix/refactoring here though.

**Comment by mavasani** [comment]
We can also consider offering 2 fixes here: “Mark member static” and “Mark member static and fix refererences”. First one is exactly the PR above, fast and unsafe with an additional warning annotation about user needing to manually fix references. The second one being the slower, safe one.

I also feel this is one of the most popular code quality rules, so should be ported to an in-box IDE analyzer/fixer. Tagging @roslyn-analysis for views.

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-analysis

**Comment by Neme12** [comment]
A code fix that doesn't fix references would still be useful. "Perfect is the enemy of good" :smile:

**Comment by mavasani** [comment]
> A local code fix that doesn't fix references would still be useful. "Perfect is the enemy of good"

Agreed. I personally don't have any problem with a code fix that leads to explicit break - my only objection is to code fix that silently changes semantics without any explicit break.
